tst/functional/TestKeepsGoingUp.kt

```diff
@@ -73,12 +82,40 @@ class TestKeepsGoingUp {
         val map = builder.build()
 
         val player = GeneralPlayer(actor)
-        val game = Game(player, map)
+        val condition: (Move) -> InvalidMoveException? =
+                { m: Move ->
+                    if (!(Math.abs(m.vector.deltax) <= 1 && Math.abs(m.vector.deltay) <= 1))  InvalidVectorException() else null
+                }
+        val game = Game(player, map, listOf(condition))
         game.run()
 
         assertEquals(exitPosition, map.positionFor(actor))
         assertEquals(true, actor.alive)
     }
+
+    @Test
+    fun testDoesntWinWithGameRulesBeingViolated() {
```

Comment:
Interesting test. Instead of customizing the rules, I'd customize the player to perform a teleporting move instead.